### ISSUE TITLE
the v3.0 should build against .net 7 and .net 8 also

### ISSUE THREAD

**Issue by ahmed-abdelrazek** [MahApps/MahApps.Metro#4408]

**Comment by tr33s** [comment]
Bump on this one

**Comment by batzen** [comment]
Why should it build against those?
What would be the benefit?

**Comment by ahmed-abdelrazek** [comment]
1- .net 8 is going to be the latest and the long term supported version of .net
2- .net 6 support going to end next year
3- new dot net features and improvements + wpf specific like file dialog
4- save other devs time (doing workarounds and hacks to solve conflicts between different versions of a dependency coming from different .net versions)
5- mah app is always behind when it comes to this type of stuff
6- why not?! i figured the library is preparing for version 3 and it would be nice if it released with .net 8 ready

**Comment by batzen** [comment]
None of your listed points requires building against a new runtime version.

I never experienced point 4 from your list.
Could you provide an example where workarounds are needed?

**Comment by ahmed-abdelrazek** [comment]
> None of your listed points requires building against a new runtime version.
what do you mean?!

it happens to me often in projects where wpf (mahapps) razor and webview2 is used together mostly because of https://www.nuget.org/packages/Microsoft.CSharp/ sometimes others too

**Comment by ahmed-abdelrazek** [comment]
also this might help
https://learn.microsoft.com/en-us/dotnet/core/compatibility/7.0
https://learn.microsoft.com/en-us/dotnet/core/compatibility/8.0

**Comment by batzen** [comment]
There is still no benefit in compiling against additional frameworks, as MahApps.Metro makes no use of deprecated or newer APIs.
Adding additional frameworks only increases the size of the package and lengthens compile times during development of MahApps.Metro.

What would make sense is to remove deprecated/out of support frameworks and add the lowest newer supported version.

Which would mean:
- Removing .NET 3.1 and 5.0
- .NET Framework 4.7 and 4.8 could also be removed as there is nothing that being used from those frameworks and 4.6.2 is enough.

v3.0 already has a target for .NET 6.0. I don't know why adding .NET 7.0 would change anything.

**Comment by ahmed-abdelrazek** [comment]
i thought .net7 support was going to end in 2025 but it's going to end even before .net6 in 2024

**Comment by Montago** [comment]
bump ... 

Currently MahApps is holding me back on 2 projects that i cant upgrade to .NET 7 or .NET 8 for that matter.

**Comment by batzen** [comment]
@Montago That's not true. Nothing in MahApps.Metro is holding you back. MahApps.Metro has .NET 5 as a target framework which works flawlessly when you target higher .NET versions.

**Comment by Montago** [comment]
> @Montago That's not true. Nothing in MahApps.Metro is holding you back. MahApps.Metro has .NET 5 as a target framework which works flawlessly when you target higher .NET versions.

I have a copy of the repo in my projects and have remove .NET5 and Core3 build target and added .NET6 (i think) 
.NET 5 give a ton of errors / warnings 

but you are right, with .NET 6 build Target, it's possible to upgrade to .NET 8 .. apparantly

**Comment by batzen** [comment]
@Montago Which warnings did you get when compiling your app while using the nuget version of MahApps.Metro?

**Comment by sonomirco** [comment]
@batzen 
I need to use the net7-windows framework, but once I use it, mahApps is not compiled. If I switch to framework48, there is no issue.
The plug-in I am developing needs to work for both frameworks. Is there a way to force MahApps to compile in Net7 or higher?

**Comment by batzen** [comment]
@sonomirco That depends on what you want to ship/do. Those folders are usually for localization/resource/satellite assemblies. MahApps.Metro itself should only produce a "de" folder.
Do the other folders also contain dlls from MahApps.Metro?

**Comment by sonomirco** [comment]
This is the list.

runtimes\win\lib\net7.0\Microsoft.Win32.SystemEvents.dll
runtimes\browser\lib\net7.0\System.Text.Encodings.Web.dll
fr-CA\UnitsNet.resources.dll
nb-NO\UnitsNet.resources.dll
ru-RU\UnitsNet.resources.dll
zh-CN\UnitsNet.resources.dll
de\MahApps.Metro.resources.dll

**Comment by batzen** [comment]
Those DLLs are not related to MahApps.Metro, but a different component you seem to reference.
